### PR TITLE
Feature/kb 12 datacatalog dump

### PR DESCRIPTION
**Pull request by parijke**

### CODE REVIEW

File: src/Application/Dto/DistributionDto.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+declare(strict_types=1);
+
+namespace LinkedDataSets\Application\Dto;
+final class DistributionDto
+{
+    private string $format;
+    private string $fileName;
+    private int $id;
+    public function __construct($format, $fileName, $id)
```
Comment (MKodde) [inline src/Application/Dto/DistributionDto.php:11]: I'd say, add type definition to the parameters of your construction. That ensures the correct data is loaded and fails early when it is not.

File: src/Application/Job/CatalogDumpJob.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        if (!$this->logger) {
+            throw new ServiceNotFoundException('The logger service is not found');
+        }
+        $this->serverUrl = $serviceLocator->get('ViewHelperManager')?->get('ServerUrl');
+        if (!$this->serverUrl) {
+            throw new ServiceNotFoundException('The serverUrl service is not found');
+        }
+        $this->id = $this->getArg('id');
+        if (!$this->id) {
+            throw new InvalidArgumentException('No catalog_id was provided to the job');
```
Comment (MKodde) [inline src/Application/Job/CatalogDumpJob.php:44]: The exception message implies the required argument is named `catalog_id` where it actually is `id`. Consider renaming the argument or the exception message.

File: src/Application/Job/CatalogDumpJob.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    protected function dumpSerialisedFiles(Graph $graph): void // candidate for separate class
+    {
+        $fileName = "datacatalog-{$this->id}"; // in separate class make a const FILENAME_PREFIX or so
+        foreach (self::DUMP_FORMATS as $format => $extension) {
+            $content = $graph->serialise($format);
+            file_put_contents(OMEKA_PATH . "/files/datacatalogs/{$fileName}." . $extension, $content);
+            $this->logger->notice(
+                "The file {$fileName}.{$extension} is available." // @translate
```
Comment (MKodde) [inline src/Application/Job/CatalogDumpJob.php:128]: I recon we do not need to translate the log entries? But if the PM specified this differently, consider my feedback nullified.

File: src/Application/Job/DataDumpJob.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        mkdir($path);
+        return $path;
+    }
+
+    private function getIdFromPath($uri): int {
+        $path = parse_url($uri, PHP_URL_PATH);
+        $segments = explode('/', $path);
+        $id = end($segments);
+        return (int) $id;
+    }
```
Comment (MKodde) [inline src/Application/Job/DataDumpJob.php:147]: Given the POC status of this initial code I understand this code is in this class. But please consider moving it away to a helper service that is injected on this Job. But feel free to implement this at a later stage.

**Review by MKodde — COMMENTED** [review]
Nice work. Great to see some of the logic come together in this PR

See some feedback in the comments below. And a more generic comment would be to choose a coding style and implement it throughout the code base. I now see different strategies mixed together. I suggest copying the one Omeka S uses and enforce it on your own module code via PHP CS.

File: src/Application/Job/DataDumpJob.php [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        fclose($handle);
+
+        $graph = new \EasyRdf\Graph();
+        $graph->parseFile($output_file);
+
+        $endFile = OMEKA_PATH . '/files/datadumps/'.$distribution->getFilename();
+        file_put_contents($endFile, $graph->serialise("jsonld"));
+
+        $size = (new \SplFileInfo($endFile))->getSize();
+
```
[the remark this answers is not in the document]
Comment (parijke) [reply src/Application/Job/DataDumpJob.php:118]: @MKodde @coret Good point.. That depends.... if the PO wants to reuse it if nothings has changed. Suppose he decides another distribution format (creates a distribution) you don't need to crawl everything again.

But you could also convert an existing distribution I guess.... Let's discuss

**Pull request merged**